File: corporate-reporting/src/types/Risk.ts
import { IEntity } from "./Entity";
import { IUser } from "./User";
import { IRiskRiskType } from "./RiskRiskType";
import { IDirectorate } from "./Directorate";
import { IProject } from "./Project";
import { IReportingEntity, ReportingEntity } from "./ReportingEntity";
import { DateService } from "../services";
import { IGroup } from "./Group";
import { IRiskRiskMitigationAction } from "./RiskRiskMitigationAction";

export interface IRisk extends IReportingEntity {
    RiskCode: string;
    GroupID: number;
    DirectorateID: number;
    ProjectID: number;
    RiskOwnerUserID: number;
    RiskRegisterID: number;
    ReportApproverUserID?: number;
    RiskEventDescription: string;
    RiskCauseDescription: string;
    RiskImpactDescription: string;
    UnmitigatedRiskProbabilityID: number;
    UnmitigatedRiskImpactLevelID: number;
    TargetRiskProbabilityID: number;
    TargetRiskImpactLevelID: number;
    RiskAppetiteID: number; // Obsolete, now calculated from other selections?
    RiskIsOngoing: boolean;
    RiskProximity: Date;
    IsProjectRisk: boolean;
    CreatedDate: Date;
    RiskRegisteredDate: Date;
    Group?: IGroup;
    Directorate?: IDirectorate;
    Project?: IProject;
    RiskRegister?: IEntity;
    RiskOwnerUser?: IUser;
    ReportApproverUser?: IUser;
    RiskRiskTypes?: IRiskRiskType[];
    UnmitigatedRiskProbability?: IEntity;
    UnmitigatedRiskImpactLevel?: IEntity;
    TargetRiskProbability?: IEntity;
    TargetRiskImpactLevel?: IEntity;
    RiskAppetite?: IEntity;
    LinkedRiskID?: number;
    ChildRisks?: IRisk[];
    RiskRiskMitigationActions?: IRiskRiskMitigationAction[];
}

export class Risk extends ReportingEntity implements IRisk {
    public RiskCode = '';
    public GroupID = null;
    public DirectorateID = null;
    public ProjectID = null;
    public RiskOwnerUserID = null;
    public ReportApproverUserID = null;
    public RiskRegisterID = null;
    public RiskEventDescription = '';
    public RiskCauseDescription = '';
    public RiskImpactDescription = '';
    public UnmitigatedRiskProbabilityID = null;
    public UnmitigatedRiskImpactLevelID = null;
    public TargetRiskProbabilityID = null;
    public TargetRiskImpactLevelID = null;
    public RiskAppetiteID = null;
    public RiskIsOngoing = false;
    public RiskProximity = null;
    public IsProjectRisk = null;
    public CreatedDate = null;
    public RiskRegisteredDate = DateService.setLocaleDateToUTC(new Date());

    public Group = null;
    public Directorate = null;
    public Project = null;
    public RiskRegister = null;
    public RiskOwnerUser = null;
    public RiskRiskTypes = [];
    public UnmitigatedRiskProbability = null;
    public UnmitigatedRiskImpactLevel = null;
    public TargetRiskProbability = null;
    public TargetRiskImpactLevel = null;
    public RiskAppetite = null;
    public LinkedRiskID = null;
    public ChildRisks = [];
    public EntityStatus = null;
    public Contributors = [];
}

File: ORB.Core/Models/Risk.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class Risk : ReportingEntity
    {
        public Risk()
        {
            SignOffs = new HashSet<SignOff>();
        }

        public string RiskCode { get; set; }
        public int? GroupID { get; set; }
        public int? DirectorateID { get; set; }
        public int? RiskOwnerUserID { get; set; }
        public int? RagOptionID { get; set; }
        public int? RiskRegisterID { get; set; }
        public string RiskEventDescription { get; set; }
        public string RiskCauseDescription { get; set; }
        public string RiskImpactDescription { get; set; }
        public int? UnmitigatedRiskProbabilityID { get; set; }
        public int? UnmitigatedRiskImpactLevelID { get; set; }
        public int? TargetRiskProbabilityID { get; set; }
        public int? TargetRiskImpactLevelID { get; set; }
        public int? RiskAppetiteID { get; set; }
        public bool? IsProjectRisk { get; set; }
        public int? ProjectID { get; set; }
        public DateTime? RiskProximity { get; set; }
        public bool? RiskIsOngoing { get; set; }
        public int? LinkedRiskID { get; set; }
        public int? ReportApproverUserID { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? RiskRegisteredDate { get; set; }

        public virtual Group Group { get; set; }
        public virtual Directorate Directorate { get; set; }
        public virtual Project Project { get; set; }
        public virtual RagOption RagOption { get; set; }
        public virtual User ReportApproverUser { get; set; }
        public virtual RiskAppetite RiskAppetite { get; set; }
        public virtual User RiskOwnerUser { get; set; }
        public virtual RiskRegister RiskRegister { get; set; }
        public virtual RiskImpactLevel TargetRiskImpactLevel { get; set; }
        public virtual RiskProbability TargetRiskProbability { get; set; }
        public virtual RiskImpactLevel UnmitigatedRiskImpactLevel { get; set; }
        public virtual RiskProbability UnmitigatedRiskProbability { get; set; }
        public virtual ICollection<SignOff> SignOffs { get; set; }
    }
}
